src/Backups/BackupEntriesCollector.cpp

```diff
@@ -101,13 +108,19 @@ BackupEntriesCollector::BackupEntriesCollector(
         context->getSettingsRef().backup_restore_keeper_max_retries,
         context->getSettingsRef().backup_restore_keeper_retry_initial_backoff_ms,
         context->getSettingsRef().backup_restore_keeper_retry_max_backoff_ms)
+    , threadpool(threadpool_)
 {
 }
 
-BackupEntriesCollector::~BackupEntriesCollector() = default;
+BackupEntriesCollector::~BackupEntriesCollector()
+{
+    threadpool.wait();
```

Comment:
This is not excessive because `BackupsWorker` is already doing this work in `BackupsWorker::shutdown()` and because of it's allowed to make multiple backups at the same time (usually we don't do that but it's allowed).